src/shared.ts

```diff
@@ -645,16 +646,31 @@ export async function BlockBlueVerified(user: BlueBlockerUser, config: Config) {
 			return;
 		}
 
+		// Step 0: Check for disallowed words or emojis in usernames.
+		const disallowedWordsWithoutEmptyStrings = config.disallowedWords.filter(word => word !== '');
+		if (disallowedWordsWithoutEmptyStrings.length > 0){
+			// this makes extra sure that emojis are always detected, regardless if they are attached to a word or another string of emojis.
+			// the 'i' makes the test case insensitive, which helps users not have to worry about typing the same word multiple times with different variations.
+			const disallowedWordsAndEmojis = new RegExp(`${config.disallowedWords.join('(?= |$)|')}(?= |$)`, 'i');
```

Comment:
I'm noticing that in trying to make partial words not have false positives (eg. "maga" matching "magazine") I kind of mangled detecting emojis anywhere in the string, since there's now a positive lookahead for a space or EOL.

Uhhhh, I don't really know what the best way to handle emojis specially would be? Maybe:
```ts
const disallowedWordsAndEmojis = config.disallowedWords.reduce(
  (regex, value) => regex + value + (/*check if value is an emoji*/) ? '|' : '(?= |$)|',
  ''
)
```

Reply:
pinging @MaxIsJoe, lmk if you have any suggestions here

Reply:
We could run another check specifically for emojis, and we don't need to be specific on where they are placed or what pattern they're in.
you can find emojis by using the regex `/\p{Emoji}/u;`, filter all the emojis inside `config.disallowedWords` then run a simpler regex test along side `disallowedWordsAndEmojis` to see if any blacklisted emoji is mentioned anywhere.

Reply:
maybe
```js
const regexTokenRegex = /[\[\]\\\.\|\{\}\(\)\?\:\<\>\*\+\'\"\=\^\$]/gi; // this feels psychotic
let str = '[test]'
console.log(str.replace(regexTokenRegex, m => '\\' + m[0]))
\[test\]
```

Reply:
my brain hurts

Reply:
I think this follows through on all of your suggestions? https://github.com/kheina-com/Blue-Blocker/pull/267/commits/480ef4e16bc59f870e043b5852e5343331732f77